helm/node-exporter-app/templates/_resource.tpl

```diff
@@ -7,6 +7,11 @@ suffix appended, e.g. "-957c9d6ff-pkzgw". Given that Kubernetes allows 63
 characters for resource names, the stem is truncated to 47 characters to leave
 room for such suffix.
 */}}
+
+{{- define "resource.daemonset.name" -}}
+{{- printf "%s-%s" .Release.Name .Chart.AppVersion | replace "." "-" | trunc 47 | trimSuffix "-" -}}
```

Comment:
Same as `kube-state-metrics` this causes the Daemonset to be recreated which allows us to upgrade to a version with the upgraded `selector` labels.